hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestBasicSchemaEvolution.scala

```diff
@@ -169,20 +169,35 @@ class TestBasicSchemaEvolution extends HoodieSparkClientTestBase with ScalaAsser
     // 2. Write 2d batch with another schema (added column `age`)
     //
 
-    val secondSchema = StructType(
+    val secondInputSchema = StructType(
```

Comment:
shouldn't the behavior be the same when reconcile is enabled?

Reply:
Yes, but reconcile is not always enabled